### PR TITLE
refactor(ShardClient): enhance shard ID retrieval and global shard count logic.

### PR DESCRIPTION
**Pull request by pryzmian**
- Enhanced `getShardId() `method with optional chaining and ternary operator
- Improved `getTotalShardCount()` method logic to account for global shard count accurately

### CODE REVIEW

File: src/core/ShardClient.ts [hunk 1]
```diff
@@ -50,12 +52,11 @@ export class ShardClient implements IShardClient {
             this._logger.error(error, 'An error occurred while connecting to the Discord gateway.');
             this._logger.warn('Make sure the DISCORD_BOT_TOKEN environment variable is set and valid.');
         }
-
         this._logger.debug('Successfully started shard client.');
     }
 
     public registerEventListener(eventName: string, once: boolean, listener: () => void): void {
-        this._logger.debug(`Registering ShardClient event listener for '${eventName}' event...`);
```
Comment (mariusbegby) [inline src/core/ShardClient.ts:58]: Keep the ShardClient in this log. It's specifically for registering events on the sharding client, so we should keep that description in the logs.

File: src/core/ShardClient.ts [hunk 2]
```diff
@@ -9,6 +9,7 @@ export class ShardClient implements IShardClient {
     private _shardClientConfig: ShardClientConfig;
     private _shardClient: Client;
     private _applicationId: string;
+    private _maxShards: ShardClientConfig['maxShards'];
```
Comment (mariusbegby) [inline src/core/ShardClient.ts:12]: This looks good

**Review by mariusbegby — APPROVED** [review]
Fix

**Comment by mariusbegby** [comment]
I am actually not sure what is correct in terms of getTotalShardCount().

We have getTotalShardCount(), and getShardCount(). The throught is that the getTotalShardCount() retrieves global shard count, across clusters. We don't have a shardmanager yet or that logic. I am really not sure how Eris handles this stuff.

The shardClient is the "local" one, which is a client towards Discord API, but can have multiple shards. However this all runs in one process, so we need a customer sharding manager to spawn multiple shard clients. Discord.js did this very simply with a builtin shard manager, but I don't think Eris has that.

This is just my assumption on how the sharding with Eris works, but I am really not sure. We can revisit this when we implement a sharding manager.

File: src/core/ShardClient.ts (hunk 1 above)
Comment (pryzmian) [reply src/core/ShardClient.ts:58]: i'll add that back

**Pull request merged**